### PR TITLE
#37: Renamed errorCodeConfig.yml to error_code_config.yml

### LINKED ISSUE #37: Change casing of errorCodeConfig.yml
Rename errorCodeConfig.yml --> error_code_config.yml

Rationale: Unify casing for project files

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #37

### CODE REVIEW

File: doc/changes/changes_0.4.0.md [hunk 1]
```diff
@@ -2,9 +2,14 @@
 
 Code name:
 
+## Summary
+
+In this release we renamed to `errorCodeConfig.yml` to `error_code_config.yml` in order to unify the casing of out project files. Don't forget to update the file when updating.
```
Comment (AnastasiiaSergienko) [inline doc/changes/changes_0.4.0.md:7]:
```suggestion
In this release we renamed `errorCodeConfig.yml` to `error_code_config.yml` in order to unify the casing of our project files. Don't forget to update the file when updating the plugin version.
```

File: README.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ This will add the plugin to the maven `verify` lifecycle phase.
 
 Each project has an individual error-tag (e.g. `EXM`). Different sub packages within the project can have different sub tags (`EXM-MODEL`). This plugin validates that the classes of the project only use the corresponding short tag.
 
-For that you need to configure the projects error-tag mapping in the `errorCodeConfig.yml` file stored in your projects root directory.
+For that you need to configure the projects error-tag mapping in the `error_code_config.yml` file stored in your projects root directory.
```
Comment (AnastasiiaSergienko) [inline README.md:50]:
```suggestion
For that you need to configure the project's error-tag mapping in the `error_code_config.yml` file stored in your project's root directory.
```

File: doc/changes/changes_0.4.0.md [hunk 3]
```diff
@@ -2,9 +2,14 @@
...
+
 ## Refactoring
 
-* #35 Remove hardcoded version numbers from tests
+* #35: Remove hardcoded version numbers from tests
```
Comment (AnastasiiaSergienko) [inline doc/changes/changes_0.4.0.md:11]:
```suggestion
* #35: Removed hardcoded version numbers from tests
```

File: doc/changes/changes_0.4.0.md [hunk 4]
```diff
@@ -2,9 +2,14 @@
...
+* #37: Renamed errorCodeConfig.yml to error_code_config.yml
```
Comment (AnastasiiaSergienko) [inline doc/changes/changes_0.4.0.md:12]:
```suggestion
* #37: Renamed `errorCodeConfig.yml` to `error_code_config.yml`
```

File: src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigException.java [hunk 5]
```diff
@@ -1,7 +1,7 @@
 package com.exasol.errorcodecrawlermavenplugin.config;
 
 /**
- * Exception that is thrown on an invalid errorCodeConfig.yml
+ * Exception that is thrown on an invalid error_code_config.yml
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigException.java:4]: I would remove the file name from the Javadoc, so you don't need to touch the comments when you rename/ change the file (as it happened now)

Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigException.java:4]: The only place I would leave the file name is the constant CONFIG_NAME

File: src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java [hunk 6]
```diff
@@ -26,7 +26,7 @@ public ErrorCodeConfigReader(final Path projectDir) throws ErrorCodeConfigExcept
         this.errorConfigFile = projectDir.resolve(CONFIG_NAME).toFile();
         if (!this.errorConfigFile.exists()) {
             throw new ErrorCodeConfigException(messageBuilder("E-ECM-9")
-                    .message("Could not find errorCodeConfig.yml in the current project.")
+                    .message("Could not find error_code_config.yml in the current project.")
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java:29]: Use CONFIG_NAME constant here

File: src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java [hunk 7]
```diff
@@ -46,7 +46,7 @@ public ErrorCodeConfig read() throws ErrorCodeConfigException {
             return mapper.readValue(this.errorConfigFile, ErrorCodeConfig.class);
         } catch (final IOException exception) {
             throw new ErrorCodeConfigException(
-                    messageBuilder("E-ECM-7").message("Failed to read projects errorCodeConfig.yml.").toString(),
+                    messageBuilder("E-ECM-7").message("Failed to read projects error_code_config.yml.").toString(),
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java:49]: And here

File: src/test/java/com/exasol/errorcodecrawlermavenplugin/ErrorCodeCrawlerMojoIT.java [hunk 8]
```diff
@@ -84,8 +84,8 @@ void beforeEach() throws IOException {
                 this.projectDir.resolve("pom.xml"), StandardCopyOption.REPLACE_EXISTING);
         Files.copy(
                 Objects.requireNonNull(
-                        getClass().getClassLoader().getResourceAsStream("testProject/errorCodeConfig.yml")),
-                this.projectDir.resolve("errorCodeConfig.yml"), StandardCopyOption.REPLACE_EXISTING);
+                        getClass().getClassLoader().getResourceAsStream("testProject/error_code_config.yml")),
+                this.projectDir.resolve("error_code_config.yml"), StandardCopyOption.REPLACE_EXISTING);
```
Comment (AnastasiiaSergienko) [inline src/test/java/com/exasol/errorcodecrawlermavenplugin/ErrorCodeCrawlerMojoIT.java:88]: And in the test too, It's public anyway

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java [hunk 9]
```diff
@@ -26,7 +26,7 @@ public ErrorCodeConfigReader(final Path projectDir) throws ErrorCodeConfigExcept
         this.errorConfigFile = projectDir.resolve(CONFIG_NAME).toFile();
         if (!this.errorConfigFile.exists()) {
             throw new ErrorCodeConfigException(messageBuilder("E-ECM-9")
-                    .message("Could not find errorCodeConfig.yml in the current project.")
+                    .message("Could not find " + CONFIG_NAME + " in the current project.")
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java:29]: Does it work like this? I always used a parameter for it :o

**Review by AnastasiiaSergienko — APPROVED** [review]
Approved with one question

File: src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java (hunk 9 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/errorcodecrawlermavenplugin/config/ErrorCodeConfigReader.java:29]: yes. But only for constants. For variables you have to use parameters

**Pull request merged**